src/db12/benchmark.py

```diff
@@ -0,0 +1,165 @@
+"""Contains all the functions needed to run DB12"""
+
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+import os
+import sys
+import random
+import multiprocessing
+import json
+
+if sys.version_info[0] < 3:
+    #pylint: disable = E, W, R, C
+    range = xrange
+
+def single_dirac_benchmark(iterations_num=1, measured_copies=None):
+    """Get Normalized Power of one CPU in DIRAC Benchmark 2012 units (DB12)"""
+    #pylint: disable = too-many-locals
+    # This number of iterations corresponds to 1kHS2k.seconds, i.e. 250 HS06 seconds
+
+    iters = int(1000 * 1000 * 12.5)
+    calib = 250.0
+    if sys.version_info[0] < 3:
+        #pylint: disable = E, W, R, C
+        m_1 = long(0)
+        m_2 = long(0)
+    else:
+        m_1 = int(0)
+        m_2 = int(0)
+    p_1 = 0
+    p_2 = 0
+    # Do one iteration extra to allow CPUs with variable speed (we ignore zeroth iteration)
+    # Do one or more extra iterations to avoid tail effects when copies run in parallel
+    it_1 = 0
+    while (it_1 <= iterations_num) or (
+            measured_copies is not None and measured_copies.value > 0
+    ):
+        if it_1 == 1:
+            start = os.times()
+
+        # Now the iterations
+        for _j in range(iters):
+            t_1 = random.normalvariate(10, 1)
+            m_1 += t_1
+            m_2 += t_1 * t_1
+            p_1 += t_1
+            p_2 += t_1 * t_1
+
+        if it_1 == iterations_num:
+            end = os.times()
+            if measured_copies is not None:
+                # Reduce the total of running copies by one
+                measured_copies.value -= 1
+
+        it_1 += 1
+
+    cput = sum(end[:4]) - sum(start[:4])
+    wall = end[4] - start[4]
+
+    if not cput:
+        return None
+
+    # Return DIRAC-compatible values
+    output = {
+        "CPU": cput,
+        "WALL": wall,
+        "NORM": calib * iterations_num / cput,
+        "UNIT": "DB12",
+    }
+    with open('output.txt', 'w') as outfile:
+        json.dump(output, outfile)
+    return output
```

Comment:
1. This should go in `__main__.py` in a specific function such as:

```python
def dumpAsJson(filename):
    with open(filename, 'w') as outfile:
        json.dump(output, outfile)
```

So that you don't need to write these lines for each function of `db12`
This function should be called by "CLI functions" (called by the subparsers) when the `--json` flag is used.
2.  
```suggestion
    with open('output.json', 'w') as outfile:
        json.dump(output, outfile)
    return output
```

3. The `--json` should be an `argparse` argument and should take a filename as input, and you can store it in `output.json` by default (you can do that with `argparse` normally).